### PR TITLE
Improve test coverage Bio.Geo

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1288 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Improving some coverage.

### CODE REVIEW

**Comment by peterjc** [comment]
Is ``str(record)`` intended to give a valid record which would be put in a file and parsed? That might make an even more useful test - can we read our own output (e.g. via ``StringIO`` rather than a temp file)?

File: Tests/test_geo.py [hunk 1]
```diff
@@ -3165,6 +3164,91 @@ def test_soft_ex_affy_chp(self):
...
+                                          "Sample_author: Donna,G,Albertson\n\n"
+                                          "Sample_description: Coriell Cell Repositories "
+                                          "cell line <a h\n"
+                                          "ref=\"http://locus.umdnj.edu/nigms/nigms_cgi/display.cgi?GM05296\">GM05296</a>.\n\n"
+                                          "Sample_description: Fibroblast cell line derived "
+                                          "from a 1 mo\n"
+                                          "nth old female with multiple congenital malformations, "
+                                          "dysmorphic features, intr\n"
+                                          "auterine growth retardation, heart murmur, cleft palate, equinovarus deformity, \n"
+                                          "microcephaly, coloboma of right iris, clinodactyly, reduced RBC catalase activit\n"
```
Comment (peterjc) [inline Tests/test_geo.py]: This line is overly long - indenting everything to the opening bracket of ``self.assertEqual(`` would be better (or adopt the ``black`` indentation style which would give even less indentation)

File: Tests/test_geo.py [hunk 2]
```diff
@@ -3165,6 +3164,91 @@ def test_soft_ex_affy_chp(self):
...
+                                          "12: 12				0\t\n"
+                                          "13: 13	0.108454	1.078072	0.005196	3\t\n"
+                                          "14: 14	0.078633	1.056017	0.009165	3\t\n"
+                                          "15: 15	0.098571	1.070712	0.007834	3\t\n"
+                                          "16: 16	0.044048	1.031003	0.013651	3\t\n"
+                                          "17: 17	0.018039	1.012582	0.005471	3\t\n"
+                                          "18: 18	-0.088807	0.9403	0.010571	3\t\n"
+                                          "19: 19	0.016349	1.011397	0.007113	3\t\n"
+                                          "20: 20	0.030977	1.021704	0.016798	3\t\n")
+
```
Comment (peterjc) [inline Tests/test_geo.py]: You would still need the ``\t`` for tabs, but would triple quotes make it easier to express this without the explicit ``\n`` for new lines - and perhaps avoid any indentation at all?

**Comment by svalqui** [comment]
I think str(record) is only to print(record), it trims the number of rows to 20.
https://github.com/biopython/biopython/blob/dd117f4366b572ed43c54b7b98a1eca0424958e0/Bio/Geo/Record.py#L69-71

File: Tests/test_geo.py [hunk 3]
```diff
@@ -3214,7 +3214,8 @@ def test_record_str(self):
 
 Sample_description: Fibroblast cell line derived from a 1 mo
 nth old female with multiple congenital malformations, dysmorphic features, intr
-auterine growth retardation, heart murmur, cleft palate, equinovarus deformity,
+auterine growth retardation, heart murmur, cleft palate, equinovarus deformity, """
+                                          """
```
Comment (peterjc) [inline Tests/test_geo.py]: These triple quotes look like an accidental edit?

Comment (svalqui) [reply Tests/test_geo.py]: Amended, the test was passing but flake8 failing, so had to insert /t or /n, that one was left behind.

**Comment by peterjc** [comment]
Thank you

**Pull request merged**